blueflood-core/src/integration-test/java/com/rackspacecloud/blueflood/service/MetricsIntegrationTest.java

```diff
@@ -162,392 +150,82 @@ public void testRollupGenerationSimple() throws Exception {
             BasicRollup basicRollup = BasicRollup.buildRollupFromRollups(input);
             writes.add(new SingleRollupWriteContext(basicRollup,
                     locator,
                     Granularity.MIN_240.coarser(),
                     CassandraModel.getColumnFamily(BasicRollup.class, Granularity.MIN_240.coarser()),
                     range.start));
         }
-        writer.insertRollups(writes);
+        basicMetricsRW.insertRollups(writes);
 
         // verify the number of points in 48h worth of rollups. 
         Range range = new Range(Granularity.MIN_1440.snapMillis(baseMillis), Granularity.MIN_1440.snapMillis(endMillis + Granularity.MIN_1440.milliseconds()));
-        Points<BasicRollup> input = reader.getDataToRoll(BasicRollup.class, locator, range,
-                CassandraModel.getColumnFamily(BasicRollup.class, Granularity.MIN_1440));
+        Points<BasicRollup> input = basicMetricsRW.getDataToRollup(locator, RollupType.BF_BASIC, range,
+                CassandraModel.getBasicColumnFamilyName(Granularity.MIN_1440));
         BasicRollup basicRollup = BasicRollup.buildRollupFromRollups(input);
         Assert.assertEquals(60 * hours, basicRollup.getCount());
     }
 
     @Test
     public void testSimpleInsertAndGet() throws Exception {
-        AstyanaxWriter writer = AstyanaxWriter.getInstance();
-        AstyanaxReader reader = AstyanaxReader.getInstance();
-        final long baseMillis = 1333635148000L; // some point during 5 April 2012.
-        long lastMillis = baseMillis + (300 * 1000); // 300 seconds.
-        final String acctId = "ac" + IntegrationTestBase.randString(8);
-        final String metricName = "fooService,barServer," + randString(8);
-
-        final Locator locator  = Locator.createLocatorFromPathComponents(acctId, metricName);
-
-        Set<Long> expectedTimestamps = new HashSet<Long>();
-        // insert something every 30s for 5 mins.
-        for (int i = 0; i < 10; i++) {
-            final long curMillis = baseMillis + (i * 30000); // 30 seconds later.
-            expectedTimestamps.add(curMillis);
-            List<Metric> metrics = new ArrayList<Metric>();
-            metrics.add(getRandomIntMetric(locator, curMillis));
-            writer.insertFull(metrics);
-        }
-        
-        Set<Long> actualTimestamps = new HashSet<Long>();
-        // get back the cols that were written from start to stop.
-
-        Points<SimpleNumber> points = reader.getDataToRoll(SimpleNumber.class, locator, new Range(baseMillis, lastMillis),
-                CassandraModel.getColumnFamily(BasicRollup.class, Granularity.FULL));
-        actualTimestamps = points.getPoints().keySet();
-        Assert.assertEquals(expectedTimestamps, actualTimestamps);
-    }
-
-    @Test
-    //In this test, string metrics are configured to be always dropped. So they are not persisted at all.
-    public void testStringMetricsIfSoConfiguredAreAlwaysDropped() throws Exception {
```

Comment:
These Strings and Boolean tests are moved to DBasicMetricsRWIntegrationTest.java and ABasicMetricsRWIntegrationTest.java, for Datastax and Astyanax respectively.